File: Observer-Pattern/Weather-Station-App/ForecastDisplay.ts
import { Observer, WeatherDisplayElement } from "../Interfaces";

export class ForecastDisplay implements Observer, WeatherDisplayElement{  
    private temperature; 
    private humidity; 
    private pressure; 
    update(temperature:number, humidity: number, pressure: number){ 
        this.temperature = temperature;
        this.pressure = pressure; 
        this.humidity = humidity; 
    } 
        display(){  
            console.log("-----FORECAST DISPLAY-----");
            console.log("Temperature is:",this.temperature);  
            console.log("Humidity is:",this.humidity);  
            console.log("Pressure is:",this.pressure); 
        }   
}

File: Java/src/main/java/ObserverPattern/ForecastDisplay.java
package ObserverPattern;

public class ForecastDisplay extends WeatherStationDisplay{
    public int temperature;
    public int pressure;
    public int humidity;

    ForecastDisplay(){
        this.temperature = 0;
        this.pressure = 0;
        this.humidity = 0;
    }
    @Override
    public void update(int temperature, int pressure, int humidity) {
        this.temperature = temperature;
        this.pressure = pressure;
        this.humidity = humidity;
    }

    public void showForecast(){
        System.out.println("Forecast is: " + this.temperature*this.humidity*this.pressure
        );
    }
}
